src/components/HomePage/EditGameBoyModal.tsx

```diff
@@ -0,0 +1,321 @@
+    title: string | undefined;
+    data: { gameId: string | undefined; description: string | undefined }[];
+  }>({
+    title: undefined,
+    data: [
+      {
+        gameId: undefined,
+        description: undefined,
+      },
+      {
+        gameId: undefined,
+        description: undefined,
+      },
+      {
+        gameId: undefined,
+        description: undefined,
+      },
+    ],
+  });
+
+  const { data: games } = useQuery({
+    queryKey: ["games"],
+    queryFn: async () => {
+      const response = await fetch("/api/games");
+      if (!response.ok) {
+        throw new Error("Failed to fetch games");
+      }
+      const data = await response.json();
+      return data.games as (IGame & { _id: string })[];
+    },
+    retry: 3,
+  });
+
+  const editGameBoy = useMutation({
+    mutationFn: () => {
+      return axios.put(
+        `/api/homepage/`,
+        JSON.stringify({
+          gameBoyTitle: newTitle,
+          gameBoys: newData,
+        }),
+        {
+          headers: {
+            "Content-Type": "application/text",
+          },
+        },
+      );
+    },
+    onSuccess: async () => {
+      await refetchHomePage();
+      setOpen(false);
+    },
+  });
+
+  function clearValidationErrors() {
+    setValidationErrors({
+      title: undefined,
+      data: [
+        {
+          gameId: undefined,
+          description: undefined,
+        },
+        {
+          gameId: undefined,
+          description: undefined,
+        },
+        {
+          gameId: undefined,
+          description: undefined,
+        },
+      ],
+    });
+  }
+
+  function validateForm() {
+    // missing title
+    if (!newTitle) {
+      setValidationErrors({
+        title: "Title is required!",
+        data: validationErrors.data,
+      });
+      return false;
+    }
+
+    // missing game 1 fields
+    if (!newData[0].gameId || !newData[0].description) {
+      setValidationErrors({
+        title: validationErrors.title,
+        data: [
+          {
+            gameId: !newData[0].gameId ? "Game is required!" : undefined,
+            description: !newData[0].description
+              ? "Description is required!"
+              : undefined,
+          },
+          validationErrors.data[1],
+          validationErrors.data[2],
+        ],
+      });
+      return false;
+    }
+
+    // missing description for other games
+    for (let i = 1; i < 3; i++) {
+      if (newData[i].gameId && !newData[i].description) {
+        setValidationErrors({
+          title: validationErrors.title,
+          data: [
+            validationErrors.data[0],
+            {
+              gameId: undefined,
+              description: !newData[i].description
+                ? "Description is required!"
+                : undefined,
+            },
+            validationErrors.data[2],
+          ],
+        });
+        return false;
+      }
+    }
+
+    // game 3 without game 2
```

Comment:
Instead of throwing an error when they fill out game 3 instead of game 2, would it be possible to just display game 3 and game 1 as two games on the home page?

Reply:
yup, just added